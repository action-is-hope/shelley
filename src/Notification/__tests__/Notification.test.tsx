import { Notification } from "..";
import renderer from "react-test-renderer";

describe("Notification", () => {
  it("success notifcation", () => {
    const tree = renderer
      .create(
        <Notification
          tone="success"
          title="Success notification title"
          subtitle="Subtitle goes here"
          data-id="inline-success-notification"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("info notifcation", () => {
    const tree = renderer
      .create(
        <Notification
          tone="info"
          title="Info notification title"
          subtitle="Subtitle goes here"
          data-id="inline-info-notification"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("warning notifcation", () => {
    const tree = renderer
      .create(
        <Notification
          tone="warning"
          title="Warning notification title"
          subtitle="Subtitle goes here"
          data-id="inline-warning-notification"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("alert notifcation", () => {
    const tree = renderer
      .create(
        <Notification
          tone="alert"
          title="Alert notification title"
          subtitle="Subtitle goes here"
          data-id="inline-alert-notification"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("isDismissable", () => {
    const tree = renderer
      .create(
        <Notification
          isDismissable
          title="Notification title"
          subtitle="Subtitle goes here"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with children", () => {
    const tree = renderer
      .create(
        <Notification title="Notification title">
          <p>Notification content goes here</p>
        </Notification>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with footer content", () => {
    const tree = renderer
      .create(
        <Notification
          title="Notification title"
          footer={<span>Do not show this again [x]</span>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
