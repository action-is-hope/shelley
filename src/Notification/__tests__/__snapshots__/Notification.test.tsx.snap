// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification alert notifcation 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-5-alert"
  data-id="inline-alert-notification"
  role="alert"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
        data-id="inline-alert-notification--title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          data-id="inline-alert-notification--icon"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM9 13h-2v-2h2v2zM9 10h-2v-7h2v7z"
          />
        </svg>
        Alert notification title
        <span
          className="Notification__Test__subtitle"
          data-id="inline-alert-notification--subTitle"
        >
          Subtitle goes here
        </span>
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      data-id="inline-alert-notification--closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        data-id="inline-alert-notification--icon"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Notification hideCloseButton 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-4-info"
  role="info"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 1c3.9 0 7 3.1 7 7s-3.1 7-7 7-7-3.1-7-7 3.1-7 7-7zM8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8v0z"
          />
          <path
            d="M7 6h2v7h-2v-7z"
          />
          <path
            d="M7 3h2v2h-2v-2z"
          />
        </svg>
        Notification title
        <span
          className="Notification__Test__subtitle"
        >
          Subtitle goes here
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Notification info notifcation 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-4-info"
  data-id="inline-info-notification"
  role="info"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
        data-id="inline-info-notification--title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          data-id="inline-info-notification--icon"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 1c3.9 0 7 3.1 7 7s-3.1 7-7 7-7-3.1-7-7 3.1-7 7-7zM8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8v0z"
          />
          <path
            d="M7 6h2v7h-2v-7z"
          />
          <path
            d="M7 3h2v2h-2v-2z"
          />
        </svg>
        Info notification title
        <span
          className="Notification__Test__subtitle"
          data-id="inline-info-notification--subTitle"
        >
          Subtitle goes here
        </span>
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      data-id="inline-info-notification--closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        data-id="inline-info-notification--icon"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Notification renders with children 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-4-info"
  role="info"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 1c3.9 0 7 3.1 7 7s-3.1 7-7 7-7-3.1-7-7 3.1-7 7-7zM8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8v0z"
          />
          <path
            d="M7 6h2v7h-2v-7z"
          />
          <path
            d="M7 3h2v2h-2v-2z"
          />
        </svg>
        Notification title
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
  <div
    className="Notification__Test__children"
  >
    <p>
      Notification content goes here
    </p>
  </div>
</div>
`;

exports[`Notification renders with footer content 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-4-info"
  role="info"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 1c3.9 0 7 3.1 7 7s-3.1 7-7 7-7-3.1-7-7 3.1-7 7-7zM8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8v0z"
          />
          <path
            d="M7 6h2v7h-2v-7z"
          />
          <path
            d="M7 3h2v2h-2v-2z"
          />
        </svg>
        Notification title
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
  <div
    className="Notification__Test__footer"
  >
    <span>
      Do not show this again [x]
    </span>
  </div>
</div>
`;

exports[`Notification success notifcation 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-7-success"
  data-id="inline-success-notification"
  role="success"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
        data-id="inline-success-notification--title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          data-id="inline-success-notification--icon"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 0c-4.4 0-8 3.6-8 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zM7.1 11.7l-4.2-4.1 1.4-1.4 2.7 2.7 5-4.9 1.4 1.4-6.3 6.3z"
          />
        </svg>
        Success notification title
        <span
          className="Notification__Test__subtitle"
          data-id="inline-success-notification--subTitle"
        >
          Subtitle goes here
        </span>
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      data-id="inline-success-notification--closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        data-id="inline-success-notification--icon"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Notification warning notifcation 1`] = `
<div
  className="Notification__Test__root Notification__Test---role-7-warning"
  data-id="inline-warning-notification"
  role="warning"
>
  <div
    className="Notification__Test__header"
  >
    <div
      className="Notification__Test__textWrapper"
    >
      <span
        className="Text__Test__root Text__Test---vol-1-3 Text__Test--hasAdornment Notification__Test__title"
        data-id="inline-warning-notification--title"
      >
        <svg
          aria-hidden={true}
          className="Icon__Test__root"
          data-id="inline-warning-notification--icon"
          viewBox="0 0 16 16"
        >
          <path
            d="M8 1l-8 14h16l-8-14zM8 13c-0.6 0-1-0.4-1-1s0.4-1 1-1 1 0.4 1 1c0 0.6-0.4 1-1 1zM7 10v-4h2v4h-2z"
          />
        </svg>
        Warning notification title
        <span
          className="Notification__Test__subtitle"
          data-id="inline-warning-notification--subTitle"
        >
          Subtitle goes here
        </span>
      </span>
    </div>
    <button
      aria-label="Close"
      className="ButtonBase__Test__root ButtonBase__Test---vol-1-2 IconButton__Test__root Notification__Test__closeButton"
      data-id="inline-warning-notification--closeButton"
      onBlur={[Function]}
      onClick={[Function]}
      onDragStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
       
      <svg
        aria-hidden={true}
        className="Icon__Test__root"
        data-id="inline-warning-notification--icon"
        viewBox="0 0 16 16"
      >
        <path
          d="M15.1 3.1l-2.2-2.2-4.9 5-4.9-5-2.2 2.2 5 4.9-5 4.9 2.2 2.2 4.9-5 4.9 5 2.2-2.2-5-4.9z"
        />
      </svg>
    </button>
  </div>
</div>
`;
